Sources/SwiftSyntaxBuilder/Trivia+Extension.swift

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension Trivia {
+  /// Creates a new `Trivia` and adds spacing if needed.
+  ///
+  /// If there is 3 or more, then we see if the first two is a newline and a space
+  /// If so, we need to replace add the correct amount of space after each newline
+  /// comming after to have the right indentation of the code.
+  func addSpacingAfterNewlinesIfNeeded() -> Trivia {
+    var updatedTriviaPieces = self.pieces
+    if updatedTriviaPieces.count > 2,
+       isNewline(trivia: updatedTriviaPieces[0]),
+       let toInset = self.makeIndentation(trivia: updatedTriviaPieces[1]) {
+
+      for i in (2..<updatedTriviaPieces.count).reversed() where isNewline(trivia: updatedTriviaPieces[i]) {
+          updatedTriviaPieces.insert(toInset, at: i + 1)
+        }
+      }
+      
+      return Trivia(pieces: updatedTriviaPieces)
+    }
+    
+    private func isNewline(trivia: TriviaPiece) -> Bool {
```

Comment:
Why not make this a computed property on `TriviaPiece`?

Reply:
wanna do as a public extension?

Reply:
I don’t have a strong preference. But making it public seems beneficial to me.